module/applications/actor/sheet-v2-mixin.mjs

```diff
@@ -260,13 +260,12 @@ export default function ActorSheetV2Mixin(Base) {
         if ( values.length ) traits.dm = values;
       }
 
-      // Display ranged communication
-      for ( const [key, { label }] of Object.entries(CONFIG.DND5E.communicationTypes) ) {
-        const data = this.actor.system.traits?.languages?.communication?.[key];
-        if ( !data?.value ) return;
-        traits.languages ??= [];
-        traits.languages.push({ label: `${label} ${formatDistance(data.value, data.units)}` });
-      }
+      // Handle languages
+      const languages = this.actor.system.traits?.languages?.labels;
+      if ( languages?.dialects?.length || languages?.ranged?.length ) traits.languages = [
+        ...languages.dialects.map(label => ({ label })),
+        ...languages.ranged.map(label => ({ label }))
```

Comment:
Probably still want to use the original loop here so telepathy can be formatted in the same way as senses and movement.